system/t06_publish/repo.py

```diff
@@ -888,3 +888,61 @@ def check(self):
         self.check_exists('public/dists/maverick/main/binary-amd64/Packages')
         self.check_exists('public/dists/maverick/main/binary-amd64/Packages.gz')
         self.check_not_exists('public/dists/maverick/main/binary-amd64/Packages.bz2')
+
+
+class PublishRepo34Test(BaseTest):
+    """
+    publish repo: skeleton files
+    """
+    fixtureCmds = [
+        "aptly repo create local-repo",
+        "aptly repo add local-repo ${files}"
+    ]
+    runCmd = "aptly publish repo -keyring=${files}/aptly.pub -secret-keyring=${files}/aptly.sec -distribution=maverick -skip-contents local-repo"
+    gold_processor = BaseTest.expand_environ
+
+    def prepare_fixture(self):
+        super(PublishRepo34Test, self).prepare_fixture()
+
+        self.write_file(os.path.join('skel', 'dists', 'maverick', 'main', 'dep11', 'README'), 'README test file')
+
+    def check(self):
+        super(PublishRepo34Test, self).check()
+
+        self.check_exists('public/dists/maverick/main/dep11/README')
+
+        self.check_exists('public/dists/maverick/Release')
+
+        readme = self.read_file('public/dists/maverick/main/dep11/README')
+        if readme != 'README test file':
+            raise Exception("README file not copied on publish")
+
```

Comment:
Add a test to confirm that skel files added with the same path as a repository file do not override the repository file. 

```suggestion
        self.write_file(os.path.join('skel', 'dists', 'maverick', 'main', 'dep11', 'README'), 'README test file')
         self.write_file(os.path.join('skel', 'dists', 'maverick', 'Release'), 'Release test file')

    def check(self):
        super(PublishRepo34Test, self).check()

        self.check_exists('public/dists/maverick/main/dep11/README')

        self.check_exists('public/dists/maverick/Release')

        readme = self.read_file('public/dists/maverick/main/dep11/README')
        if readme != 'README test file':
            raise Exception("README file not copied on publish")

        release = self.read_file('public/dists/maverick/Release')
        if release == 'Release test file':
            raise Exception("Release file was copied on publish")

```